### ISSUE TITLE
Request execution time not captured correctly by Application Insights for ASP.NET Core apps

### ISSUE THREAD

**Issue by puneet-gupta** [Microsoft/ApplicationInsights-aspnetcore#350]
Request execution time not captured correctly by Application Insights for ASP.NET Core apps

I have a basic thread.sleep call in my controller method and I am just trying to simulate a slow request using this code in the controller

```
public IActionResult Contact()
        {
            Thread.Sleep(5000);
            ViewData["Message"] = "Your contact page.";

            return View();
}

```
 
![pic1](https://cloud.githubusercontent.com/assets/8878114/23248644/69c6ec80-f9c6-11e6-94b2-94d7b613b0ea.jpg)
 don’t know why it is showing 1.2 seconds instead of 5 seconds. If I look at the TRACE telemetry events, I see 5 second delay but then I don’t understand what exactly does REQUEST signify in this context

I am using Visual Studio 2015 and here is a snippet of project.json to show the versions of packages that I am using

"dependencies": {
    "Microsoft.AspNetCore.Razor.Tools": {
      "version": "1.0.0-preview2-final",
      "type": "build"
    },
    "Microsoft.NETCore.App": {
      "version": "1.1.0",
      "type": "platform"
    },
    "BundlerMinifier.Core": "2.3.327",
    "Microsoft.AspNetCore.Mvc": "1.1.1",
    "Microsoft.AspNetCore.Server.Kestrel": "1.1.0",
    "Microsoft.AspNetCore.StaticFiles": "1.1.0",
    "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.1.0",
    "Microsoft.Extensions.Configuration.Json": "1.1.0",
    "Microsoft.Extensions.Logging": "1.1.0",
    "Microsoft.Extensions.Logging.Console": "1.1.0",
    "Microsoft.Extensions.Logging.Debug": "1.1.0",
    "Microsoft.Extensions.Options.ConfigurationExtensions": "1.1.0",
    "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.1.0",
    "Microsoft.AspNetCore.Diagnostics": "1.1.0",
    "Microsoft.AspNetCore.Routing": "1.1.0",
    "Microsoft.AspNetCore.Server.IISIntegration": "1.1.0",
    "Microsoft.AspNetCore.AzureAppServicesIntegration": "1.0.0",
    "Microsoft.ApplicationInsights.AspNetCore": "2.0.0",
    "System.Data.SqlClient": "4.1.0"
  },

**Comment by pharring** [comment]
I couldn't repro this, however I believe that we have fixed a few bugs in the 'develop' branch that might be responsible. In particular, #343 improved the way we calculate durations from the stopwatch timestamp on machines where the stopwatch frequency isn't exactly the same as the TimeSpan 'ticks'.

I'd like to point you to a pre-release NuGet package to try. Unfortunately, we haven't published one since the 2.0.0 release. If you're able to build this repo's 'develop' branch you might be able to build and test your own version.

**Comment by billpratt** [comment]
Any update on this? I am seeing similar things.

**Comment by billpratt** [comment]
We had to downgrade to 1.0 to get correct times. If this is fixed in the develop branch, can we see an update pushed soon for 2.0?

**Comment by billpratt** [comment]
What's the status of this bug? I see that it was added to the 2.1-Beta2 milestone but there has not been activity on this repo for a month and no 2.1-Beta nuget packages? I'm just trying to get an idea on when we can reliably start using 2.X version of this. I completely understand that there is a lot in flight right now for you guys. Is there anything the community can do either with PRs or testing?

@DamianEdwards @shanselman

**Comment by pharring** [comment]
Next pre-release drop of this repo is scheduled for 3/31/17
Stable on 4/14/17
Up-to-date schedule here:
https://github.com/Microsoft/ApplicationInsights-Home/wiki/SDK-Release-Schedule

**Comment by billpratt** [comment]
That's good to here! I really liked the work done for dependency tracking in 2.0 and was sad to downgrade. I'll be sure to check out the beta release on 3/31 and provide feedback.

**Comment by billpratt** [comment]
Awesome! I really liked the work done for dependency tracking in 2.0 and was sad to downgrade.

Looking over the schedule, this issue won't be resolved and available for testing until milestone

[4/21/2017 - 2.2-Beta1 ](https://github.com/Microsoft/ApplicationInsights-aspnetcore/milestone/22)

which is a month away. That is unfortunate.

**Comment by pharring** [comment]
I actually think the issue of incorrect execution times has already been fixed with #343 and the fix will be available in the 2.1 release scheduled for 4/14.
I realize this issue is currently tagged for 2.2-Beta1, but I don't have permission to change the milestone tag.
@dnduffy please can you change the milestone tag to [2.1-Beta1](https://github.com/Microsoft/ApplicationInsights-aspnetcore/milestone/21) and resolve this as "fixed by #343" ?

**Comment by dnduffy** [comment]
Fixed by #343.
